iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2333,6 +2369,26 @@ static void SubscribeToMqttProtocol(PMQTTTRANSPORT_HANDLE_DATA transport_data)
     else
     {
         transport_data->currPacketState = PUBLISH_TYPE;
+
+        // On a service reconnect, reset the expired time of messages waiting for a PUBACK
+        // this will cause the messages to republish
```

Comment:
```suggestion
        // On a service reconnect, reset the expired time of messages waiting for a PUBACK.
        // This will cause the messages to republish in order as required by the MQTT spec.
```